### PR TITLE
chore: add goimports

### PR DESCRIPTION
**Pull request by lesam**
Closes #

- [x] [Well-formatted commit messages](https://www.conventionalcommits.org/en/v1.0.0-beta.3/)
- [x] Rebased/mergeable
- [x] Tests pass

### CODE REVIEW

**Comment by lesam** [comment]
Failure before running `make fmt`

File: cmd/internal/input.go [hunk 1]
```diff
@@ -3,11 +3,12 @@ package internal
 import (
 	"errors"
 	"fmt"
-	"github.com/influxdata/influxdb/v2/task/options"
 	"os"
 	"strings"
 	"time"
 
+	"github.com/influxdata/influxdb/v2/task/options"
+
```
Comment (danxmoran) [inline cmd/internal/input.go:11]: I'm surprised it'd leave this blank line

File: http/backup_service.go [hunk 2]
```diff
@@ -11,6 +11,7 @@ import (
 
 	"github.com/influxdata/httprouter"
 	"github.com/influxdata/influxdb/v2"
+
```
Comment (danxmoran) [inline http/backup_service.go:14]: This blank line also surprises me (can we delete the commented-out line below instead?)

File: task/backend/analytical_storage.go [hunk 3]
```diff
@@ -5,9 +5,10 @@ import (
 	"context"
 	"encoding/json"
 	"fmt"
-	"github.com/influxdata/influxdb/v2/kit/errors"
 	"time"
 
+	"github.com/influxdata/influxdb/v2/kit/errors"
+
```
Comment (danxmoran) [inline task/backend/analytical_storage.go:11]: Surprising blank line

**Review by danxmoran — COMMENTED** [review]
Raging against the machine

**Comment by lesam** [comment]
It turns out `goimports` just splits std libraries to the top, and all other groups are just sorted.

Manually fixed the grouping for some files.

File: tsdb/index/tsi1/cache.go [hunk 4]
```diff
@@ -136,7 +136,7 @@ func (c *TagValueSeriesIDCache) Put(name, key, value []byte, ss *tsdb.SeriesIDSe
 
 	// No map for the measurement - first tag key for the measurement.
 	c.cache[string(name)] = map[string]map[string]*list.Element{
-		string(key): map[string]*list.Element{string(value): listElement},
```
Comment (lesam) [inline tsdb/index/tsi1/cache.go:139]: I believe this is the `gofmt -s` - I guess nobody ran a `make fmt` recently

File: etc/checkfmt.sh [hunk 5]
```diff
@@ -1,12 +1,15 @@
 #!/bin/bash
 
+go install golang.org/x/tools/cmd/goimports
+
 HAS_FMT_ERR=0
 # For every Go file in the project, excluding vendor...
-for file in $(go list -f '{{$dir := .Dir}}{{range .GoFiles}}{{printf "%s/%s\n" $dir .}}{{end}}' ./...); do
+
+for file in $(go list -f '{{$dir := .Dir}}{{range .GoFiles}}{{printf "%s/%s\n" $dir .}}{{end}}{{range .TestGoFiles}}{{printf "%s/%s\n" $dir .}}{{end}}{{range .IgnoredGoFiles}}{{printf "%s/%s\n" $dir .}}{{end}}{{range .CgoFiles}}{{printf "%s/%s\n" $dir .}}{{end}}' ./... ); do
```
Comment (lesam) [inline etc/checkfmt.sh:8]: Even more fun - we were not doing format checking on tests, cgo files, or ignored files like `tools.go`

File: etc/checkfmt.sh [hunk 6]
```diff
@@ -1,12 +1,15 @@
...
   # ... if file does not contain standard generated code comment (https://golang.org/s/generatedcode)...
   if ! grep -Exq '^// Code generated .* DO NOT EDIT\.$' $file; then
-    FMT_OUT="$(gofmt -l -d -e $file)" # gofmt exits 0 regardless of whether it's formatted.
```
Comment (danxmoran) [inline etc/checkfmt.sh:8]: Should we keep the `gofmt` call if `goimports` only checks imports?

**Review by danxmoran — COMMENTED** [review]
My GoLand linter will be very happy after this change

File: etc/checkfmt.sh (hunk 6 above)
Comment (lesam) [reply etc/checkfmt.sh:8]: `goimports` also does formatting . The only thing it is missing is the `-s` argument to `gofmt` - do we want to block PR's on that?

Comment (lesam) [reply etc/checkfmt.sh:8]: See also https://github.com/golang/go/issues/21476

Comment (lesam) [reply etc/checkfmt.sh:8]: Note we weren't blocking on `-s` before.

**Review by danxmoran — APPROVED** [review]

**Pull request merged**